src/main/java/touro/simondrums/AudioPlayer.java

```diff
@@ -0,0 +1,73 @@
+package touro.simondrums;
+
+import javax.sound.sampled.*;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.concurrent.CountDownLatch;
+
+public class AudioPlayer {
+
+    InputStream inputStream;
+    AudioInputStream audioIn;
+
+    private Clip setupAudioClip(String fileName) throws UnsupportedAudioFileException, IOException, LineUnavailableException {
+        CountDownLatch syncLatch = new CountDownLatch(1);
+        inputStream = ListenerEvents.class.getClassLoader().getResourceAsStream(fileName);
+        audioIn = AudioSystem.getAudioInputStream(inputStream);
+        Clip clip = AudioSystem.getClip();
+        clip.addLineListener(e -> {
+            if (e.getType() == LineEvent.Type.STOP) {
+                syncLatch.countDown();
+            }
+        });
+        clip.open(audioIn);
+        //TODO: Do we still need this if we have to redo this whole method every time anyway?
```

Comment:
I don't think so. Shouldn't we only need to set position if we're not starting at the beginning?